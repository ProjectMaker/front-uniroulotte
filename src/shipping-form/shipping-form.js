import React, { Component } from 'react'
import withStyle from './with-style'

class FileUploader extends Component {
  state = {
    name: 'toot'
  }

  render() {
    return (<p>{this.state.name}</p>)
  }
}
const ShippingForm = ({onChange, fields, onValid, sendingInProgress}) => {
  return (
    <div>
      <div>
        <input name="lastname" onChange={onChange}/>
        {fields.lastname.error ?
          <span className="error" id="lastnameError">Not blanck</span>
        : null}
      </div>
      <div>
        <input name="firstname" />
      </div>
      <div>
        <button disabled={sendingInProgress} onClick={onValid}/>
      </div>
    </div>
  )
}

// export default withStyle({root: { border: '1px' } })(ShippingForm)
export default ShippingForm
export { FileUploader }
