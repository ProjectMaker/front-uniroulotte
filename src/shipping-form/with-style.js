import React, {Component} from 'react'

export default (styles) => {
  return (WrappedComponent) => {
    return class extends Component {
       render() {
         return <WrappedComponent {...this.props} classes={styles} />
       }
    }
  }
}
