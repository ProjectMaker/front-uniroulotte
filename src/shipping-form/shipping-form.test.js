import React from 'react'
import { shallow, mount } from 'enzyme'

import ShippingForm from './shipping-form'
import { FileUploader } from './shipping-form'
describe('<ShippingForm />', () => {
  it('Should display state', () => {
    const wrapper = shallow(<FileUploader/>)
    console.log(wrapper.debug())
    expect(true).toBeTruthy()
  })
  /*
  it('Should render correctly', () => {
    const wrapper = shallow(shallow(<ShippingForm/>).get(0))
    expect(wrapper).toMatchSnapshot()
  })
  */
  /*
  it('Should call onChange props', () => {
    const onChange = jest.fn()
    const fields = {
      firstname: { error: 'blanck' },
      lastname: { error: 'blanck' },
      email: { error: 'blanck' },
      emailConfirm: { error: 'blanck' },
      phoneNumber: { error: 'blanck' },
    }
    const wrapper = shallow(shallow(<ShippingForm onChange={onChange} fields={fields} />).get(0))
    wrapper.find('[name="lastname"]').simulate('change',
      { target: { value: 'abc' } }
    )
    expect(wrapper.find('[id="lastnameError"]')).toHaveLength(1)
    console.log(wrapper.debug())
    expect(onChange).toHaveBeenCalledTimes(1)
  })
  */
  /*
  it('Should call onValid props', () => {
    let wrapper
    const onValid = jest.fn(async () => {
      await wrapper.setProps({sendingInProgress: true})
      expect(wrapper.find('button').props().disabled).toBeTruthy()
    })
    wrapper = shallow(shallow(<ShippingForm onValid={onValid} />).get(0))
    wrapper.find('button').simulate('click')
    expect(onValid).toHaveBeenCalled()
  })
  */
})
