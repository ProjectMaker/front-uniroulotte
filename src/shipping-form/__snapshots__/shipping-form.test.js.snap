// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingForm /> Should render correctly 1`] = `ShallowWrapper {}`;
